### ISSUE TITLE
onError function of Mutation has mismatched types

### ISSUE THREAD

**Issue by Code-Victor** [HuolalaTech/react-query-kit#48]
# Issue
In my use Mutation router, I found out the Error and Variable types are swapped.
![image](https://github.com/HuolalaTech/react-query-kit/assets/63613078/c24d3098-6381-4b37-b934-7da0e17d0cca)

## Replication
The router
```tsx
export const authRouter = router("auth", {
  verifyEmail: router.mutation({
    mutationFn: verifyEmail,
  }),
  login: router.mutation({
    mutationFn: loginUser,
  }),
  register: router.mutation({
    mutationFn: registerUser,
  }),
});
```
The function
```ts
export async function loginUser(data: { email: string; password: string }) {
  const res = await api.post<LoginUserResponse>("/login", data);
  return res.data;
}
```

**Comment by liaoliao666** [comment]
it is fixed in v3.1.3.

**Comment by Code-Victor** [comment]
Alright, thanks 👍🏾
